### PR TITLE
[linux] Allow overriding aot_library_path

### PR DESCRIPTION
**Pull request by dballard**
Support overriding `aot_library_path` in `FLDartProject` on Linux.

Fixes: https://github.com/flutter/flutter/issues/128213

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/linux/fl_dart_project.cc [hunk 1]
```diff
@@ -62,6 +62,13 @@ G_MODULE_EXPORT FlDartProject* fl_dart_project_new() {
   return self;
 }
 
+G_MODULE_EXPORT void fl_dart_project_set_aot_library_path(FlDartProject* self,
+                                                          gchar* path) {
```
Comment (robert-ancell) [inline shell/platform/linux/fl_dart_project.cc:66]: Use 'const gchar*'

File: shell/platform/linux/public/flutter_linux/fl_dart_project.h [hunk 2]
```diff
@@ -35,6 +35,16 @@ G_DECLARE_FINAL_TYPE(FlDartProject, fl_dart_project, FL, DART_PROJECT, GObject)
  */
 FlDartProject* fl_dart_project_new();
 
+/**
+ * fl_dart_project_set_aot_library_path:
+ * @project: an #FlDartProject.
```
Comment (robert-ancell) [inline shell/platform/linux/public/flutter_linux/fl_dart_project.h:40]: Missing '@path: ...'

File: shell/platform/linux/public/flutter_linux/fl_dart_project.h [hunk 3]
```diff
@@ -35,6 +35,16 @@ G_DECLARE_FINAL_TYPE(FlDartProject, fl_dart_project, FL, DART_PROJECT, GObject)
...
+ *
+ * Sets the path to the AOT library in the Flutter application, which is
+ * the path to libapp.so. By default this is lib/libapp.so relative to the
+ * executable directory.
+ */
+void fl_dart_project_set_aot_library_path(FlDartProject* project, gchar* path);
```
Comment (robert-ancell) [inline shell/platform/linux/public/flutter_linux/fl_dart_project.h:46]: Use 'const gchar*'

**Review by robert-ancell — CHANGES_REQUESTED** [review]
Seems to be missing the code to pass this path to the engine?

**Comment by loic-sharma** [comment]
@dballard Have you had a chance to address the feedback? (Desktop triage)

**Comment by dballard** [comment]
@loic-sharma Not yet, been a busy few weeks but hoping to get back to this in the next few days, thanks for the reminder!

**Comment by dballard** [comment]
> Seems to be missing the code to pass this path to the engine?

How so? I copied what `fl_dart_project_set_icu_data_path` does. Doing this from with in my own code seemed to be sufficient.

**Comment by robert-ancell** [comment]
> > Seems to be missing the code to pass this path to the engine?
> 
> How so? I copied what `fl_dart_project_set_icu_data_path` does. Doing this from with in my own code seemed to be sufficient.

Sorry, yes. I missed the existing code was already there to do this.

File: shell/platform/linux/public/flutter_linux/fl_dart_project.h [hunk 4]
```diff
@@ -35,6 +35,18 @@ G_DECLARE_FINAL_TYPE(FlDartProject, fl_dart_project, FL, DART_PROJECT, GObject)
  */
 FlDartProject* fl_dart_project_new();
 
+/**
+ * fl_dart_project_set_aot_library_path:
+ * @project: an #FlDartProject.
+ * @path: the absolute path to the assets directory.
```
Comment (robert-ancell) [inline shell/platform/linux/public/flutter_linux/fl_dart_project.h:41]: Copy/paste error.

Comment (robert-ancell) [reply shell/platform/linux/public/flutter_linux/fl_dart_project.h:41]: Other than this small fix this looks good, thanks!

**Review by robert-ancell — APPROVED** [review]
LGTM

**Pull request merged**